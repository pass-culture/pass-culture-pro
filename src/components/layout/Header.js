import { Icon } from 'pass-culture-shared'
import classnames from 'classnames'
import get from 'lodash.get'
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { NavLink } from 'react-router-dom'
import { compose } from 'redux'

import SignoutButton from './SignoutButton'
import Logo from './Logo'

class Header extends Component {
  constructor() {
    super()
    this.state = {
      showMobileMenu: false,
    }
  }

  render() {
    const { location, name, whiteHeader } = this.props
    const { search } = location
    const { showMobileMenu } = this.state
    return (
      <header className={classnames('navbar', { 'is-primary': !whiteHeader })}>
        <div className="container">
          <div className="navbar-brand">
            <Logo className="navbar-item" whiteHeader={whiteHeader} />
            <span
              className="navbar-burger"
              onClick={e =>
                this.setState({
                  showMobileMenu: !showMobileMenu,
                })
              }>
              <span />
              <span />
              <span />
            </span>
          </div>
          <div
            className={classnames('navbar-menu', {
              'is-active': showMobileMenu,
            })}>
            <div className="navbar-end">
              <NavLink className="navbar-item" to={'/guichet'} key={0}>
                <span className="icon">
                  <Icon svg={'ico-guichet-w'} />
                </span>
                <span>Guichet</span>
              </NavLink>
              {!whiteHeader && (
                <NavLink
                  className="navbar-item"
                  to={`/offres${search}`}
                  key={1}>
                  <span className="icon">
                    <Icon svg={'ico-offres-w'} />
                  </span>
                  <span>Vos offres</span>
                </NavLink>
              )}
              <div className="navbar-item has-dropdown is-hoverable">
                <NavLink className="navbar-link" to="#">
                  <span className="icon">
                    <Icon svg={`ico-user-circled${whiteHeader ? '' : '-w'}`} />
                  </span>
                  <span>{name}</span>
                </NavLink>
                <div className="navbar-dropdown is-right">
                  <NavLink to={'/profil'} className="navbar-item">
                    <span className="icon">
                      <Icon svg={'ico-user'} />
                    </span>
                    <span>Profil</span>
                  </NavLink>
                  <NavLink to={'/structures'} className="navbar-item">
                    <span className="icon">
                      <Icon svg={'ico-structure-r'} />
                    </span>
                    <span>Structures</span>
                  </NavLink>
                  {false && (
                    <NavLink to={'/delegations'} className="navbar-item">
                      <span className="icon">
                        <Icon svg={'ico-delegation-r'} />
                      </span>
                      <span>Délégations</span>
                    </NavLink>
                  )}
                  <NavLink to={'/comptabilite'} className="navbar-item">
                    <span className="icon">
                      <Icon svg={'ico-compta'} />
                    </span>
                    <span>Comptabilité</span>
                  </NavLink>
                  <SignoutButton Tag="a" className="navbar-item">
                    <span className="icon">
                      <Icon svg={'ico-deconnect'} />
                    </span>
                    <span>Déconnexion</span>
                  </SignoutButton>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
    )
  }
}

export default compose(
  withRouter,
  connect(state => ({
    name: get(state, 'user.publicName'),
  }))
)(Header)
