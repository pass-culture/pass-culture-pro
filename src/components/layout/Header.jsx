import get from 'lodash.get'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { compose } from 'redux'

import RawHeader from './RawHeader'

const mapStateToProps = state => {
  const name = get(state, 'user.publicName')
  return {
    name,
    offerers: state.data.offerers,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(RawHeader)
