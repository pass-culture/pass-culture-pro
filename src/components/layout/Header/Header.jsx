import PropTypes from 'prop-types'
import classnames from 'classnames'
import { SignoutButton } from 'pass-culture-shared'
import React, { PureComponent } from 'react'
import { NavLink } from 'react-router-dom'

import Icon from '../Icon'
import Logo from '../Logo'
import { HELP_PAGE_URL, IS_DEV } from '../../../utils/config'

class Header extends PureComponent {
  constructor() {
    super()
    this.state = {
      showMobileMenu: false,
    }
  }

  onHandleSuccessRedirect = () => '/connexion'

  handleOnClick = showMobileMenu => () => {
    this.setState({
      showMobileMenu: !showMobileMenu,
    })
  }

  render() {
    const { isSmall, name, whiteHeader, offerers } = this.props
    const { showMobileMenu } = this.state

    return (
      <header
        className={classnames('navbar', {
          'is-primary': !whiteHeader,
          'is-small': isSmall,
        })}
      >
        <div className="container">
          <div className="navbar-brand">
            <Logo
              className="navbar-item"
              whiteHeader={whiteHeader}
            />
            <button
              className="navbar-burger"
              onClick={this.handleOnClick(showMobileMenu)}
              type="button"
            >
              <span />
              <span />
              <span />
            </button>
          </div>
          <div
            className={classnames('navbar-menu', {
              'is-active': showMobileMenu,
            })}
          >
            <div className="navbar-end">
              <NavLink
                className="navbar-item"
                to="/guichet"
              >
                <span className="icon">
                  <Icon svg="ico-guichet-w" />
                </span>
                <span>
                  {'Guichet'}
                </span>
              </NavLink>
              {!whiteHeader && (
                <NavLink
                  className="navbar-item"
                  to="/offres"
                >
                  <span className="icon">
                    <Icon svg="ico-offres-w" />
                  </span>
                  <span>
                    {'Offres'}
                  </span>
                </NavLink>
              )}
              <a
                className="navbar-item"
                href={HELP_PAGE_URL}
                rel="noopener noreferrer"
                target="_blank"
              >
                <span className="icon">
                  <Icon svg="ico-help-w" />
                </span>
                <span>
                  {'Aide'}
                </span>
              </a>
              <div className="navbar-item has-dropdown is-hoverable">
                <NavLink
                  className="navbar-link"
                  to="#"
                >
                  <span className="icon">
                    <Icon svg={`ico-user-circled${whiteHeader ? '' : '-w'}`} />
                  </span>
                  <span>
                    {name}
                  </span>
                </NavLink>
                <div className="navbar-dropdown is-right">
                  <NavLink
                    className="navbar-item"
                    to="/profil"
                  >
                    <span className="icon">
                      <Icon svg="ico-user" />
                    </span>
                    <span>
                      {'Profil'}
                    </span>
                  </NavLink>
                  <NavLink
                    className="navbar-item"
                    to="/structures"
                  >
                    <span className="icon">
                      <Icon svg="ico-structure-r" />
                    </span>
                    <span>
                      {offerers.length > 1 ? 'Structures juridiques' : 'Structure juridique'}
                    </span>
                  </NavLink>
                  {false && (
                    <NavLink
                      className="navbar-item"
                      to="/delegations"
                    >
                      <span className="icon">
                        <Icon svg="ico-delegation-r" />
                      </span>
                      <span>
                        {'Délégations'}
                      </span>
                    </NavLink>
                  )}
                  <NavLink
                    className="navbar-item"
                    to="/reservations"
                  >
                    <span className="icon">
                      <Icon svg="ico-bookings" />
                    </span>
                    <span>
                      {'Réservations'}
                    </span>
                  </NavLink>
                  <NavLink
                    className="navbar-item"
                    to="/remboursements"
                  >
                    <span className="icon">
                      <Icon svg="ico-compta" />
                    </span>
                    <span>
                      {'Remboursements'}
                    </span>
                  </NavLink>
                  {false && (
                    <NavLink
                      className="navbar-item"
                      to="/comptabilite"
                    >
                      <span className="icon">
                        <Icon svg="ico-compta" />
                      </span>
                      <span>
                        {'Comptabilité'}
                      </span>
                    </NavLink>
                  )}
                  {IS_DEV && (
                    <NavLink
                      className="navbar-item"
                      to="/styleguide"
                    >
                      <span className="icon">
                        <Icon svg="ico-stars" />
                      </span>
                      <span>
                        {'Styleguide'}
                      </span>
                    </NavLink>
                  )}
                  <SignoutButton
                    className="navbar-item"
                    handleSuccessRedirect={this.onHandleSuccessRedirect}
                    Tag="a"
                  >
                    <span className="icon">
                      <Icon svg="ico-deconnect" />
                    </span>
                    <span>
                      {'Déconnexion'}
                    </span>
                  </SignoutButton>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
    )
  }
}

Header.defaultProps = {
  isSmall: false,
  whiteHeader: false,
}

Header.propTypes = {
  isSmall: PropTypes.bool,
  name: PropTypes.string.isRequired,
  offerers: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  whiteHeader: PropTypes.bool,
}

export default Header
