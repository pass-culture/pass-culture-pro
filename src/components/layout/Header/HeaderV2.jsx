import React from 'react'
import { useDispatch } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { reinitializeData } from 'redux-saga-data'

import { signout } from '../../../repository/pcapi/pcapi'
import Logo from '../Logo'

import { ReactComponent as BookingsSvg } from './assets/bookings.svg'
import { ReactComponent as CounterSvg } from './assets/counter.svg'
import { ReactComponent as HomeSvg } from './assets/home.svg'
import { ReactComponent as OffersSvg } from './assets/offers.svg'
import { ReactComponent as RefundsSvg } from './assets/refunds.svg'
import { ReactComponent as SignoutSvg } from './assets/signout.svg'

const HeaderV2 = () => {
  const dispatch = useDispatch()

  function onSignoutClick() {
    signout().then(() => {
      dispatch(reinitializeData())
    })
  }

  return (
    <header className="menu-v2">
      <nav>
        <div className="nav-brand">
          <Logo className="nav-item" />
        </div>

        <div className="nav-menu">
          <NavLink
            className="nav-item"
            role="menuitem"
            to="/accueil"
          >
            <HomeSvg />
            <span>
              {'Accueil'}
            </span>
          </NavLink>

          <NavLink
            className="nav-item"
            role="menuitem"
            to="/guichet"
          >
            <CounterSvg />
            <span>
              {'Guichet'}
            </span>
          </NavLink>

          <NavLink
            className="nav-item"
            role="menuitem"
            to="/offres"
          >
            <OffersSvg />
            <span>
              {'Offres'}
            </span>
          </NavLink>

          <NavLink
            className="nav-item"
            role="menuitem"
            to="/reservations"
          >
            <BookingsSvg />
            <span>
              {'RÃ©servations'}
            </span>
          </NavLink>

          <NavLink
            className="nav-item"
            role="menuitem"
            to="/remboursements"
          >
            <RefundsSvg />
            <span>
              {'Remboursements'}
            </span>
          </NavLink>

          <div className="separator" />

          <button
            className="nav-item icon-only"
            onClick={onSignoutClick}
            role="menuitem"
            type="button"
          >
            <SignoutSvg />
          </button>
        </div>
      </nav>
    </header>
  )
}

export default HeaderV2
