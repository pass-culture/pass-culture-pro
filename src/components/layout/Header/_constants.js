const ENV_NAME = process.env.ENVIRONMENT_NAME

export const HELP_PAGE_URL = 'https://aide.passculture.app/fr/category/acteurs-culturels-1t20dhs/'
export const STYLEGUIDE_ACTIVE = ENV_NAME !== 'production' && ENV_NAME !== 'staging'
