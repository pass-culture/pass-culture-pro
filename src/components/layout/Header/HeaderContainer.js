import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { compose } from 'redux'

import Header from './Header'

export const mapStateToProps = state => {
  const { user, data } = state
  const { publicName: name } = user
  const { offerers } = data

  return {
    name,
    offerers,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(Header)
