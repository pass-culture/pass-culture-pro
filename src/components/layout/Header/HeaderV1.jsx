import { Menu, MenuButton, MenuItem, MenuLink, MenuList } from '@reach/menu-button'
import '@reach/menu-button/styles.css'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { NavLink } from 'react-router-dom'
import { reinitializeData } from 'redux-saga-data'

import { signout } from 'repository/pcapi/pcapi'

import Icon from '../Icon'
import Logo from '../Logo'

import { HELP_PAGE_URL, STYLEGUIDE_ACTIVE } from './_constants'
import { ReactComponent as BookingsSvg } from './assets/bookings.svg'
import { ReactComponent as CounterSvg } from './assets/counter.svg'
import { ReactComponent as OffersSvg } from './assets/offers.svg'
import { ReactComponent as RefundsSvg } from './assets/refunds.svg'

// FIXME (Adrien S, 11/12/2020): Inject StyleGuide as a props, to avoid mocks and increase testability

class HeaderV1 extends PureComponent {
  onHandleSuccessRedirect = () => '/connexion'

  onSignoutClick = () => {
    const { dispatch } = this.props
    signout().then(() => {
      dispatch(reinitializeData())
    })
  }

  render() {
    const { name, offerers } = this.props

    return (
      <header className="menu">
        <nav>
          <div className="nav-brand">
            <Logo className="nav-item" />
          </div>
          <div className="nav-menu">
            <NavLink
              className="nav-item"
              role="menuitem"
              to="/guichet"
            >
              <span className="icon">
                <CounterSvg />
              </span>
              <span>
                {'Guichet'}
              </span>
            </NavLink>
            <NavLink
              className="nav-item"
              role="menuitem"
              to="/offres"
            >
              <span className="icon">
                <OffersSvg />
              </span>
              <span>
                {'Offres'}
              </span>
            </NavLink>
            <NavLink
              className="nav-item"
              role="menuitem"
              to="/reservations"
            >
              <span className="icon">
                <BookingsSvg />
              </span>
              <span>
                {'Réservations'}
              </span>
            </NavLink>
            <Menu>
              <MenuButton className="nav-item">
                <span className="icon">
                  <Icon
                    height="28"
                    svg="ico-user-circled-w"
                  />
                </span>
                <span>
                  {name}
                </span>
                <Icon svg="ico-arrow-down-r" />
              </MenuButton>
              <MenuList>
                <MenuLink
                  as="a"
                  href="/profil"
                >
                  <span className="icon">
                    <Icon svg="ico-user" />
                  </span>
                  <span>
                    {'Profil'}
                  </span>
                </MenuLink>
                <MenuLink
                  as="a"
                  href="/structures"
                >
                  <span className="icon">
                    <Icon svg="ico-structure-r" />
                  </span>
                  <span>
                    {offerers.length > 1 ? 'Structures juridiques' : 'Structure juridique'}
                  </span>
                </MenuLink>
                <MenuLink
                  as="a"
                  href="/remboursements"
                >
                  <span className="icon">
                    <RefundsSvg className="primary-color" />
                  </span>
                  <span>
                    {'Remboursements'}
                  </span>
                </MenuLink>
                {STYLEGUIDE_ACTIVE && (
                  <MenuLink
                    as="a"
                    href="/styleguide"
                  >
                    <span className="icon">
                      <Icon svg="ico-stars" />
                    </span>
                    <span>
                      {'Styleguide'}
                    </span>
                  </MenuLink>
                )}
                <MenuLink
                  as="a"
                  href={HELP_PAGE_URL}
                  target="_blank"
                >
                  <span className="icon">
                    <Icon svg="ico-help" />
                  </span>
                  {'Aide'}
                </MenuLink>
                <MenuItem onSelect={this.onSignoutClick}>
                  <span className="icon">
                    <Icon svg="ico-deconnect" />
                  </span>
                  <span>
                    {'Déconnexion'}
                  </span>
                </MenuItem>
              </MenuList>
            </Menu>
          </div>
        </nav>
      </header>
    )
  }
}

HeaderV1.propTypes = {
  dispatch: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  offerers: PropTypes.arrayOf(PropTypes.shape()).isRequired,
}

export default HeaderV1
