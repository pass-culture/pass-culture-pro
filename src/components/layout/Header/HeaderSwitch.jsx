import React from 'react'

import { STYLEGUIDE_ACTIVE } from './_constants'
import HeaderV1 from './HeaderV1'
import HeaderV2 from './HeaderV2'

const HeaderSwitch = ({ isNewHomepageActive, isUserAdmin, ...headerV1Props }) => {
  if (isNewHomepageActive) {
    return (
      <HeaderV2
        isStyleguideActive={STYLEGUIDE_ACTIVE}
        isUserAdmin={isUserAdmin}
      />
    )
  } else {
    return <HeaderV1 {...headerV1Props} />
  }
}

export default HeaderSwitch
