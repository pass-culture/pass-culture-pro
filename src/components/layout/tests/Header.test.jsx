import React from 'react'
import { shallow } from 'enzyme'

import Header from '../Header'

describe('src | components | Layout | Header', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given

      // when
      const wrapper = shallow(<Header />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
