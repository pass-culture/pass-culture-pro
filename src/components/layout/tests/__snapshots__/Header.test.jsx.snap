// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | Layout | Header snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Connect(Header)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
